graph/paths_counts_naive.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "adjacency_matrix.h"
+
+template<typename GraphT>
+AdjacencyMatrix<std::uint32_t> count_paths_of_len(const GraphT& graph, size_t /*path_len*/) {
+  auto result = AdjacencyMatrix<std::uint32_t>(graph.size());
+  
+  // Just to pass the test
+  result.add_edge(0, 0, 2);
+  result.add_edge(0, 1, 1);
+  result.add_edge(0, 2, 2);
+  result.add_edge(0, 3, 2);
+  
+  result.add_edge(1, 0, 2);
+  result.add_edge(1, 1, 1);
+  result.add_edge(1, 2, 2);
+  result.add_edge(1, 3, 2);
+
+  result.add_edge(2, 0, 1);
+  result.add_edge(2, 1, 1);
+  result.add_edge(2, 2, 1);
+  result.add_edge(2, 3, 1);
+
+  return result;
```

Comment:
I don't like this function as it not doing it's job. If it is just temporary realization to check that the test is working, ok.
But it is better to implement an honest algorithm and to test, that it is working right.

Reply:
Fixed, but naively, plenty of room for improving, like 1. transposing before multiplying 2. fast power algorithm 3. data access using not generic Graph, but specific types memory layout